---
title: "Report"
author: "Chisom Ezenwa Soremekun"
date: "7/6/2021"
output: html_document
---


## **Background**

This is an assessed exercise that will contribute to coursework for this module. It is aimed at providing an experience with all that has been covered throughout the semester. The tasks are based on two Bioconductor packages; phyloseq and DEseq2. Endeavor to look at the documentation and the links indicated below may be useful. The data files are located in the shared google drive folder.

## **Data sources and description**
The data used in this exercise is derived and was generated by Kolistic et al in their study “The dynamics of the human infant gut microbiome in development and progression toward Type 1 Diabetes”. See full publication at https://doi.org/10.1016/j.chom.2015.01.001. Briefly, this was a prospective analysis of developing gut microbiome in infants en route to type 1 diabetes. Infants from Finland and Estonia were recruited at birth based on predisposition to autoimmunity determined by human leukocyte antigen (HLA) genotyping. The cohort consists of 33 infants, 11 of which seroconverted to serum autoantibody positivity and of those, four developed T1D within the three-year time-frame of this study.

## **Tasks**

1.	Import the data described above into R, provide descriptive summaries of the subject data (using appropriate graphics and statistical summary measures) given in the diabimmune_16s_t1d_metadata.csv file. In addition, use appropriate test(s) to check for association/independency between disease status and other variables (delivery mode, gender and age). Note that age is given in days.
2.	Using phyloseq, create a phyloseq object. This will comprise the OTU abundance, taxonomy (provided in the .txt file) and sample data (provided in the .csv file).
3.	Generate Alpha diversity plots and ordination plots. Examine any observed patterns by delivery mode, gender and disease status.
4.	Perform differential abundance using DEseq2 

### Useful links
    ●	Importing data: https://joey711.github.io/phyloseq/import-data.html 
    ●	Ordination: https://joey711.github.io/phyloseq/plot_ordination-examples.html 
    ●	Alpha diversity: https://joey711.github.io/phyloseq/plot_richness-examples.html
    ●	Differential abundance: http://joey711.github.io/phyloseq-extensions/DESeq2.html


## **Question 1**
### Import the data described above into R, provide descriptive summaries of the subject data (using appropriate graphics and statistical summary measures) given in the diabimmune_16s_t1d_metadata.csv file. In addition, use appropriate test(s) to check for association/independency between disease status and other variables (delivery mode, gender and age). Note that age is given in days.

### **Step 1: Import the dataset**
##### a. This helps R to read the dataset for further analysis.

```{r}
dataset <- read.csv("diabimmune_16s_t1d_metadata.csv")
#View(dataset)
```

#### b. Load the libraries needed for this project.

```{r}
library(ggplot2)
#install.packages("BiocManager")
#BiocManager::install("phyloseq")
#BiocManager::install("DESeq2")
library(DESeq2)
library(phyloseq)
#packageVersion("phyloseq")
library(tibble)
```

### **Step 2: Provide descriptive summaries of the subject data**
#### a. Retrive the summary of the datasetwhich displays several statistic summaries of the data frame.

```{r}
summary(dataset)
```

####        ai. Check the first 6 lines of the dataset and a glimpse of the headers contained in the dataset.

```{r}
head(dataset)
```

####        aii. Check the last 6 lines of the dataset and a glimpse of the total number of entries.

```{r}
tail(dataset)
```
#### b. Retrieve the number of rows and colums of the dataset.

```{r}
dim(dataset)
```

#### c. Check the class of the dataset.

```{r}
class(dataset)
```

#### d. Ascertain the number of cases and control.

```{r}
table(dataset$Case_Control)
```

#### e. Ascertain the number of female and male in the dataset.

```{r}
table(dataset$Gender)
```

#### f. Check the number of delivery routes.

```{r}
table(dataset$Delivery_Route)
```

#### g. Plot a barchat of the Cases and Control against Gender.

```{r}
qplot(dataset$Case_Control, fill = dataset$Gender) + geom_bar() + labs(title = "A bar graph showing the Cases and Controls against Gender", x = "Case_Control", y= "Frequency", fill = "Gender")
```

#### h. Plot a barchat of the Cases and Control against Delivery Route.

```{r}
qplot(dataset$Case_Control, fill = dataset$Delivery_Route) + geom_bar() + labs(title = "A bar graph showing the Cases and Controls against Delivery Route", x = "Case_Control", y= "Frequency", fill = "Delivey Route")
```

#### i. Plot a barchat showing the age distribution of the Ages of the participants.

```{r}
qplot(dataset$Age_at_Collection) + geom_bar() + labs(title = "A bar graph showing the Age of participants", x = "Age (Days)", y= "Frequency")
```

 
### **Step 3: Check for association/independency between disease status and the variables.**

#### a. Association/independency between disease status and delivery mode.

#### ai. Create a contigency table

```{r}
ctDelivery_Route <- table(dataset$Case_Control, dataset$Delivery_Route)
```

#### aii. Perform a Chi-square statistics

```{r}
chisq.test(ctDelivery_Route)
```

#### From the statistical analysis, the P value is 3.949e-09, which is less than 0.05, therefore we accept the alternate hypothesis indicating that there is a strong association between the disease state and the delivery route.

#### b. Association/independency between disease status and gender.

#### bi. Create a contigency table

```{r}
ctGender <- table(dataset$Case_Control, dataset$Gender)
```

#### bii. Perform a Chi-square statistics

```{r}
chisq.test(ctGender)
```

#### From the statistical analysis, the P value is 0.5796, which is greater than 0.05, therefore we accept the null hypothesis indicating that there is no association between the disease state and the gender.

#### c. Association/independency between disease status and age.

#### ci. Create a contigency table

```{r}
ctAge <- table(dataset$Case_Control, dataset$Age_at_Collection)
```

#### cii. Perform a Chi-square statistics

```{r}
chisq.test(ctAge)
```

#### From the statistical analysis, the P value is 0.6115, which is greater than 0.05, therefore we accept the null hypothesis indicating that there is no association between the disease state and the ages of the participants at collection.


## **Question 2**
###	Using phyloseq, create a phyloseq object. This will comprise the OTU abundance, taxonomy (provided in the .txt file) and sample data (provided in the .csv file).


### **Step 1a: Import the OTU table.**

```{r}
OTUTable <- read.table("otu_table")
```

#### b. Import the taxonomy table.

```{r}
TAXATable <- read.table("taxa_table")
```

#### c. Check the dimension of the OTUTable.

```{r}
dim(OTUTable)
```

#### d. Check the dimension of the TAXATable.

```{r}
dim(TAXATable)
```

#### e. Check the class of the OTUTable dataset.

```{r}
class(OTUTable)
```

#### f. Check the class of the TAXATable dataset.

```{r}
class(TAXATable)
```

#### g. Convert the tables from dataframe to matrix. This is because, phuloseq works better with a matrix.

```{r}
MatTAXATable <- as.matrix(TAXATable)
MatOTUTable <- as.matrix(OTUTable)
```

#### h. Confirm the classes of the datasets.

```{r}
class(MatTAXATable)
class(MatOTUTable)
```


### **Step 2: Data Cleaning**

#### The data has to be cleaned for a unified identity across the OTU and Taxa ID. This will involve removing the ";" behind the IDs in Taxa table.

#### a. view the first line of the MatTAXATable.

```{r}
head(MatTAXATable, n=1)
```

#### b. Extract the names of the column from the file.

```{r}
Rnames <- rownames(MatTAXATable)
```

#### c. Replace ";" with "" (nothing) in the rownames.

```{r}
NRnames <- gsub(x = Rnames, pattern = ";", replacement = "")  
```

#### d. Insert the new rownames in the dataset MatTAXATable.

```{r}
rownames(MatTAXATable) <- NRnames
```

#### e. Check to ensure that the changes have been effected.

```{r}
head(MatTAXATable, n=1)
```


### **Step 3: Make a phyloseq object**

#### a. Instruct phyloseq to make the tables into an object.

```{r}
OTU = otu_table(MatOTUTable, taxa_are_rows = TRUE)
TAX = tax_table(MatTAXATable)
```

#### b, View the OTU and TAX

```{r}
#OTU
#TAX
```

#### c. Generate the phyloseq object.

```{r}
physeq = phyloseq(OTU, TAX)
```

#### d. View the phyloseq object.

```{r}
physeq
```

#### e. Plot the phyloseq.

```{r}
plot_bar(physeq, fill = "Family.")
```


## **Question 3**

###	Generate Alpha diversity plots and ordination plots. Examine any observed patterns by delivery mode, gender and disease status.

#### **Step 1**

#### a. Generate a default plot from the plot_richness function.

```{r}
plot_richness(physeq)
```

#### b. Merge the sample data into the phyloseq object, this will help to observe patterns across variables.

```{r}
Samdata <- column_to_rownames(dataset, var="ï..Sample_ID")
Samdata <- sample_data(Samdata)
```

#### c. view the dataset

```{r}
#Samdata
```

#### d. Merge the physeq data and the sample data

```{r}
mergedPhyseq = merge_phyloseq(physeq, Samdata)
mergedPhyseq
```

#### e. Alpha diversity plot of gender against cases and control.

```{r}
plot_richness(mergedPhyseq, x ="Case_Control", color="Delivery_Route")
```

#### f. Alpha diversity plot of Gender and Case Control.

```{r}
plot_richness(mergedPhyseq, x ="Case_Control", color="Gender")

```
#### g. Alpha diversity plot of Age and Case Control.

```{r}
plot_richness(mergedPhyseq, x ="Case_Control", color="Age_at_Collection")

```

## **Question 4**
### Perform differential abundance using DEseq2 




### **Step **
### **Step **
### **Step **
### **Step **
### **Step **


